packages/main/src/plugin/extension/extension-loader.ts

```diff
@@ -1646,9 +1635,14 @@ export class ExtensionLoader {
       extensionUri,
       secrets,
     };
-    let deactivateFunction = undefined;
-    if (typeof extensionMain?.['deactivate'] === 'function') {
-      deactivateFunction = extensionMain['deactivate'];
+    let deactivateFunction: (() => Promise<void>) | undefined = undefined;
+    if (
+      extensionMain &&
+      'deactivate' in extensionMain &&
+      typeof extensionMain?.['deactivate'] === 'function' &&
+      isAsyncFunction(extensionMain?.['deactivate'])
```

Comment:
May I ask why we need to check that the function is async? we could just cast it to a promise, and it would not raise any error

<img width="268" height="92" alt="Image" src="https://github.com/user-attachments/assets/36145fc0-096b-455c-9e7e-faeda147f89a" />